pkg/webhook/core.oam.dev/v1alpha2/traitdefinition/validator_test.go

```diff
@@ -0,0 +1,201 @@
+		},
+		"HavePatch_OutputHasInvalidGVK_2": {
+			reason: "An error should be returned if output has no GVK",
+			template: `
+      patch: {
+       spec: replicas: parameter.replicas
+      }
+      output: {
+      	apiVersion: 123 
+      	kind:       "ManualScalerTrait"
+      	spec: {
+      	    replicaCount: parameter.replicas
+      	}
+      }`,
+			want: errors.New(errGVKRequired),
+		},
+		"HavePatch_OutputHasGVK": {
+			reason: "No error should be returned if have a patch and output has GVK",
+			template: `
+      patch: {
+       spec: replicas: parameter.replicas
+      }
+      output: {
+      	apiVersion: "core.oam.dev/v1alpha2"
+      	kind:       "ManualScalerTrait"
+      	spec: {
+      	    replicaCount: parameter.replicas
+      	}
+      }`,
+			want: nil,
+		},
+		"HavePatch_OutputHasGVK_OutputsHaveGVK": {
+			reason: "No error should be returned if have a patch and all outputs has GVK",
+			template: `
+      patch: {
+       spec: replicas: parameter.replicas
+      }
+      output: {
+      	apiVersion: "core.oam.dev/v1alpha2"
+      	kind:       "ManualScalerTrait"
+      	spec: {
+      	    replicaCount: parameter.replicas
+      	}
+      }
+      outputs: faketrait: {
+      	apiVersion: "core.oam.dev/v1alpha2"
+      	kind:       "ManualScalerTrait"
+      	spec: {
+      	    replicaCount: parameter.replicas
+      	}
+      }`,
+			want: nil,
+		},
+		"NoPatch_OutputHasGVK": {
+			reason: "No error should be returned if output has GVK",
+			template: `
+      output: {
+      	apiVersion: "core.oam.dev/v1alpha2"
+      	kind:       "ManualScalerTrait"
+      	spec: {
+      	    replicaCount: parameter.replicas
+      	}
+      }`,
+			want: nil,
+		},
+		"NoPatch_OutputHasNoGVk": {
+			reason: "An error should be returned if output has no GVK",
+			template: `
+      output: {
+      	spec: {
+      	    replicaCount: parameter.replicas
+      	}
+      }`,
+			want: errors.New(errGVKRequired),
+		},
+		"NoPatch_OutputsHaveGVK": {
+			reason: "No error should be returned if each outputs has GVK",
+			template: `
+      outputs: faketrait: {
+      	apiVersion: "core.oam.dev/v1alpha2"
+      	kind:       "ManualScalerTrait"
+      	spec: {
+      	    replicaCount: parameter.replicas
+      	}
+      }`,
+			want: nil,
+		},
+		"NoPatch_OutputHasGVK_OutputsHaveNoGVK": {
+			reason: "An error should be returned if any of outputs has no GVK",
+			template: `
+      output: {
+      	apiVersion: "core.oam.dev/v1alpha2"
+      	kind:       "ManualScalerTrait"
+      	spec: {
+      	    replicaCount: parameter.replicas
+      	}
+      }
+      outputs: faketrait: {
+      	spec: {
+      	    replicaCount: parameter.replicas
+      	}
+      }`,
+			want: errors.New(errGVKRequired),
+		},
+		"NoPatch_NoOutput": {
+			reason: "An error should be returned if definitionRef is missing",
+			template: `
+      fakefield: fakefieldvalue`,
+			want: errors.New(errDefinitionReferenceRequired),
+		},
+	}
+
```

Comment:
please add two cases with multiple `outputs`:

1. all `outputs` are all valid
2. one is invalid and the whole to be invalid

Reply:
Fixed.